packages/runtime/runtime-utils/src/summaryUtils.ts

```diff
@@ -388,6 +389,28 @@ export class TelemetryContext implements ITelemetryContext {
 		this.telemetry.set(`${prefix}${property}`, value);
 	}
 
+	/**
+	 * {@inheritDoc @fluidframework/runtime-definitions#ITelemetryContext.append}
+	 */
+	push(prefix: string, property: string, value: TelemetryEventPropertyType[]): void {
+		const prevValue = this.telemetry.get(`${prefix}${property}`);
+		if (typeof prevValue === "string") {
+			try {
+				const values = JSON.parse(prevValue) as TelemetryEventPropertyType[];
```

Comment:
This dance with continuous stringification and parsing is very expensive. Can we simply story as as an array?
The telemetry pipeline knows how to deal with arrays (that support was added some time ago, though I do not see it reflected in type system, i.e. TelemetryBaseEventPropertyType has no arrays, but take a look at convertToBasePropertyTypeUntagged() - it deals with "object" type and there is comment about arrays), so I think we should be totally fine with having here something like Record<string, TelemetryBaseEventPropertyType >[]

Even better would be to fix telemetry type system to make it clear - arrays are accepted (though need to be very careful here - they are accepted by ITelemetryLogger, but not by ITelemetryBaseLogger. I thought we split types for these two interfaces (i.e. ITelemetryLogger should accepts more types and be able to convert them), but I'm not sure I see that.

I'd recommend tracing history and pinging who was added array types (sorry, I forgot who did it) - I think this work somewhat incomplete without this type fixup.

Also adding @markfields to review / validate my train of thought